Gordon360/ApiControllers/ScheduleControlController.cs

```diff
@@ -0,0 +1,196 @@
+        {
+            _scheduleControlService = scheduleControlService;
+        }
+
+        /// <summary>
+        /// Get schedule information of local user
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("")]
+        public IHttpActionResult Get()
+        {
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var id = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "id").Value;
+
+            object scheduleControlResult = null;
+            
+            scheduleControlResult = _unitOfWork.ScheduleControlRepository.GetById(id);
+
+            if (scheduleControlResult == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(scheduleControlResult);
+
+        }
+
+
+
+        /// <summary>
+        /// Get schedule information of specific user
+        /// </summary>
+        /// <param name="username">username</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{username}")]
+        public IHttpActionResult Get(string username)
+        {
+            object scheduleControlResult = null;
+
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+            scheduleControlResult = _unitOfWork.ScheduleControlRepository.GetById(id);
+            
+            if (scheduleControlResult == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(scheduleControlResult);
+
+        }
+
+
+
+        /// <summary>
+        /// Update privacy of schedule
+        /// </summary>
+        /// <param name="value">Y or N</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("update/privacy/{value}")]
+        public IHttpActionResult UpdateSchedulePrivacy(string value)
+        {
+            // Verify Input
+            if (!ModelState.IsValid)
+            {
+                string errors = "";
+                foreach (var modelstate in ModelState.Values)
+                {
+                    foreach (var error in modelstate.Errors)
+                    {
+                        errors += "|" + error.ErrorMessage + "|" + error.Exception;
+                    }
+
+                }
+                throw new BadInputException() { ExceptionMessage = errors };
+            }
+
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var id = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "id").Value;
+            _scheduleControlService.UpdateSchedulePrivacy(id, value);
+
+            return Ok();
+
+        }
+
+        /// <summary>
+        /// Update schedule description
+        /// </summary>
+        /// <param name="value">New description</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("update/description/{value}")]
```

Comment:
Why do we put the value in the URL? It's already passed in as a parameter, right?

Also same as above, update seems unnecessary for this route.

Reply:
This is to have consistency with other codes, which requires value on url while still having on parameters 

e.g.
```
[HttpPut]
[Route("image_privacy/{value}")]
        public IHttpActionResult UpdateImagePrivacy(string value)]
```

Reply:
Change made on route